core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java

```diff
@@ -82,193 +81,197 @@ public void testLeftEquiJoin()
-                                ImmutableMap.of("ORDERS_OK", "LINEITEM_OK"),
-                                anyTree(
-                                        tableScan("orders", ImmutableMap.of("ORDERS_OK", "orderkey"))),
-                                exchange(
-                                        project(
-                                                tableScan("lineitem", ImmutableMap.of("LINEITEM_OK", "orderkey")))))));
+                        rightJoin()
+                                .withEquiCriteria(equiJoinClause("ORDERS_OK", "LINEITEM_OK"))
+                                .withDynamicFilter(ImmutableMap.of("ORDERS_OK", "LINEITEM_OK"))
+                                .withLeft(
+                                        anyTree(
+                                                tableScan("orders", ImmutableMap.of("ORDERS_OK", "orderkey"))))
+                                .withRight(
+                                        exchange(
+                                                project(
+                                                        tableScan("lineitem", ImmutableMap.of("LINEITEM_OK", "orderkey")))))
+                                .build()));
     }
 
     @Test
     public void testRightEquiJoinWithLeftExpression()
     {
         assertPlan("SELECT o.orderkey FROM orders o RIGHT JOIN lineitem l ON l.orderkey + 1 = o.orderkey",
                 output(
-                        join(
-                                RIGHT,
-                                ImmutableList.of(equiJoinClause("ORDERS_OK", "expr")),
-                                ImmutableMap.of("ORDERS_OK", "expr"),
-                                anyTree(
-                                        tableScan("orders", ImmutableMap.of("ORDERS_OK", "orderkey"))),
-                                anyTree(
-                                        project(
-                                                ImmutableMap.of("expr", expression("LINEITEM_OK + BIGINT '1'")),
-                                                tableScan("lineitem", ImmutableMap.of("LINEITEM_OK", "orderkey")))))));
+                        rightJoin()
+                                .withEquiCriteria(equiJoinClause("ORDERS_OK", "expr"))
+                                .withDynamicFilter(ImmutableMap.of("ORDERS_OK", "expr"))
+                                .withLeft(
+                                        anyTree(
+                                                tableScan("orders", ImmutableMap.of("ORDERS_OK", "orderkey"))))
+                                .withRight(
+                                        anyTree(
+                                                project(
+                                                        ImmutableMap.of("expr", expression("LINEITEM_OK + BIGINT '1'")),
+                                                        tableScan("lineitem", ImmutableMap.of("LINEITEM_OK", "orderkey")))))
+                                .build()));
```

Comment:
nit: If we could get rid of extra `build` step somehow, but I think it's fine for now